packages/api/src/routes/medical/internal-cq.ts

```diff
@@ -115,39 +155,133 @@ router.get(
     if (Config.isSandbox()) return res.sendStatus(httpStatus.NOT_IMPLEMENTED);
     const cq = makeCarequalityManagementAPI();
     if (!cq) throw new Error("Carequality API not initialized");
+    const cxId = getUUIDFrom("query", req, "cxId").orFail();
+    const facilityId = getFrom("query").optional("facilityId", req);
     const oid = getFrom("params").orFail("oid", req);
-    const getInactive = getFromQueryAsBoolean("getInactive", req);
-    const resp = await cq.listOrganizations({ count: 1, oid, active: !getInactive });
-    const org = parseCQDirectoryEntries(resp);
-
-    if (org.length > 1) {
-      const msg = "More than one organization with the same OID found in the CQ directory";
-      console.log(msg, oid);
-      capture.message(msg, {
-        extra: { context: `carequality.directory`, oid, organizations: org, level: "info" },
-      });
+
+    if (facilityId) {
+      await getFaciltiyByOidOrFail({ cxId, id: facilityId, oid });
+    } else {
+      await getOrganizationByOidOrFail({ cxId, oid });
     }
```

Comment:
We are we adding this? This would limit us from trying to get any org on the directory by oid

Reply:
These endpoints are tied to a CX, so it's not unreasonable to make sure the OID matches whatever the target CX is to ensure we're updating the correct OID. Just ensures no accidental updates (to an extent).

I've added back in the non-CX endpoint GET under `legacy/:oid`.